### PR TITLE
[FLINK-29436][build] Upgrade spotless for running on JDK 17

### PR DESCRIPTION
**Pull request by tisonkun**
This blocker is fixed by: https://github.com/diffplug/spotless/pull/1224 and https://github.com/diffplug/spotless/pull/1228.

### CODE REVIEW

File: pom.xml
Comment (tisonkun) [inline pom.xml:1035]: @zentol I read this line after submitting the patch. Does it mean that google format works differently for Java 8 and Java 17? Or just it cannot work on Java 17 so the result is "different".

Comment (zentol) [reply pom.xml:1035]: It...kinda means what it says.

The google java format we use (1.7; _not the spotless version_), does not run on Java 17.
Newer versions that do run format things slightly differently w.r.t. lambdas, but IIRC no longer run on Java 8.

**Comment by zentol** [comment]
Could there be a bug in that spotless version that the skip parameter is being ignored?

**Comment by tisonkun** [comment]
@zentol with this change I tested JDK 17 can work with spotless:

```bash
gh pr checkout 20911
mvn -nsu spotless:apply
```

... while it generates diff on:

* flink-connectors/flink-sql-connector-hive-3.1.2/src/main/java/org/apache/hadoop/hive/conf/HiveConf.java
* flink-connectors/flink-sql-connector-hive-3.1.2/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

And I pushed a second commit for the change.

**Comment by tisonkun** [comment]
It seems the version bump cause this configure not work, I will try to find a reasonable followup:

```xml
			<!-- disable spotless since we need copy hive code -->
			<plugin>
				<groupId>com.diffplug.spotless</groupId>
				<artifactId>spotless-maven-plugin</artifactId>
				<configuration>
					<skip>true</skip>
				</configuration>
			</plugin>
```

**Comment by tisonkun** [comment]
Pushed a workaround on the plugin configs. I'm submitting a patch for upstream to avoid such a workaround.

**Review by zentol — APPROVED** [review]
+1 % CI passing

**Comment by tisonkun** [comment]
@zentol thanks for your reviewing.

I've submitted a patch to the upstream https://github.com/diffplug/spotless/pull/1353. 

Supposed 1) we're not in a hurry on this change 2) it is not easily to conflict with other patches 3) upstream uses auto-release after a patch is merged. I suggest after CI passes we wait a few days (e.g., after the Chinese National Day holiday I'll come back on Oct. 8th) to try to eliminate the workaround in this patch.

**Comment by zentol** [comment]
Sure, let's wait for the release.

**Comment by tisonkun** [comment]
Upgrade to 2.27.1 and eliminate the workaround. Wait for the new CI report and merge.

**Comment by tisonkun** [comment]
@flinkbot run azure

**Comment by tisonkun** [comment]
Although test fails should be unrelated, IIRC someone said you should get CI passed to merge a patch, lol. Rerun.

**Pull request merged**
